File: summer2019/week2/intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer> set = new HashSet<>();
        for (int n : nums1) {
            set.add(n);
        }
        HashSet<Integer> resultSet = new HashSet<>();
        for (int n : nums2) {
            if (set.contains(n) && !resultSet.contains(n)) {
                resultSet.add(n);
            }
        }
        int[] result = new int[resultSet.size()];
        int p = 0;
        for (int n : resultSet) {
            result[p++] = n;
        }
        return result;
    }
}

File: summer2021/unit2/intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1Set = set(nums1)
        resultSet = set()
        for i in nums2:
            if i in nums1Set:
                resultSet.add(i)
        return list(resultSet)